### PR TITLE
Extract rx

### PR DESCRIPTION
**Pull request by zsoltk**
Extract remaining pieces of RxJava from base module + reorg packages

### CODE REVIEW

File: libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/CompositeCancellable.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.badoo.ribs.minimal.reactive
+
+class CompositeCancellable(vararg cancellables: Cancellable) : Cancellable {
+    private val items = mutableListOf(*cancellables)
+    operator fun plusAssign(item: Cancellable) {
+        items += item
```
Comment (CherryPerry) [inline libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/CompositeCancellable.kt:6]: Should we do something, when CompositeCancellable is already cancelled and we are trying to add new Cancellable to the list?
CompositeDisposable disposes automatically in this case AFAIK.

File: libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/Relay.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.badoo.ribs.minimal.reactive
+
+import com.badoo.ribs.minimal.reactive.Cancellable.Companion.cancellableOf
```
Comment (CherryPerry) [inline libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/Relay.kt:3]: Drop `minimal` package part maybe?
Do not think it has any value.

File: libraries/rib-rx2/src/main/java/com/badoo/ribs/rx2/workflows/RxWorkflowActivity.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.badoo.ribs.rx2.workflows
+
+import android.content.Intent
+import android.os.Bundle
+import com.badoo.ribs.android.RibActivity
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+
+abstract class RxWorkflowActivity : RibActivity() {
```
Comment (CherryPerry) [inline libraries/rib-rx2/src/main/java/com/badoo/ribs/rx2/workflows/RxWorkflowActivity.kt:9]: Do you think we still need this class?
I can't find any usages of `workflowFactory` in our code base.
Personally I think it does not gives clients any particular value and reason to use it. It is quite simple to check `intent` in any way you want from client code and handle just it. Also this class has some flaws like: only  `ACTION_VIEW` is supported, no `onNewIntent` handling, RxJava support only, requirement to handle the node instance manually after creation (`workflowFactory` does not provide the node).

File: libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/CompositeCancellable.kt (hunk 1 above)
Comment (zsoltk) [reply libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/CompositeCancellable.kt:6]: For me this is beyond the scope of this PR (can lead down a longer road), but feel free to open another one if you're interested.

File: libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/Relay.kt (hunk 2 above)
Comment (zsoltk) [reply libraries/rib-base/src/main/java/com/badoo/ribs/minimal/reactive/Relay.kt:3]: I'd like to keep all the reinvented wheels in one place.

File: libraries/rib-rx2/src/main/java/com/badoo/ribs/rx2/workflows/RxWorkflowActivity.kt (hunk 3 above)
Comment (zsoltk) [reply libraries/rib-rx2/src/main/java/com/badoo/ribs/rx2/workflows/RxWorkflowActivity.kt:9]: Yeah that's a fair point. Missing support can be added though. So far all of this still existed, but as part of RibActivity. Now at least it's extracted as a concern. I'd still rather keep it to provide a starting point.

**Pull request merged**